### PR TITLE
Remove affinities label from kubevirt_vmi_cpu_affinity and use sum as value

### PR DESCRIPTION
**Pull request by machadovilaca**
**What this PR does / why we need it**:

Fixes: #9713

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

jira-ticket: https://issues.redhat.com/browse/CNV-28961

**Release note**:

```release-note
Remove affinities label from kubevirt_vmi_cpu_affinity and use sum as value
```

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -186,6 +186,9 @@ The total number of tx packets dropped on vNIC interfaces. Type: Counter.
 ### kubevirt_vmi_network_transmit_packets_total
 Total network traffic transmitted packets. Type: Counter.
 
+### kubevirt_vmi_node_cpu_affinity
+VMI CPU affinities to node physical cores count. Type: Gauge.
```
Comment (sradco) [inline docs/metrics.md:190]: Perhaps it would be clearer as 
Number of VMI CPU affinities to node physical cores. Type: Gauge.

Comment (machadovilaca) [reply docs/metrics.md]: I like it, updated

File: pkg/virt-operator/resource/generate/components/prometheus.go [hunk 2]
```diff
@@ -687,6 +687,14 @@ func GetRecordingRules(namespace string) []KubevirtRecordingRule {
 			MType:       prometheusv1.MetricTypeCounter,
 			Description: "The total number of requests to deprecated KubeVirt APIs.",
 		},
+		{
```
Comment (enp0s3) [inline pkg/virt-operator/resource/generate/components/prometheus.go:690]: @machadovilaca  Sorry but I am not following the deprecation logic here, you actually state about deprecation of `kubevirt_vmi_cpu_affinity` but you also remove that metric, and make it a rule instead? can you please elaborate on the strategy?

File: tools/prom-metrics-collector/metrics_collector.go [hunk 3]
```diff
@@ -49,7 +49,6 @@ var excludedMetrics = map[string]struct{}{
 	"kubevirt_virt_operator_leading_total":                 struct{}{},
 	"kubevirt_virt_operator_ready_total":                   struct{}{},
 	"kubevirt_virt_operator_up_total":                      struct{}{},
-	"kubevirt_vmi_cpu_affinity":                            struct{}{},
```
Comment (sradco) [inline tools/prom-metrics-collector/metrics_collector.go:52]: @machadovilaca You dont need to add the new metric name here?

Comment (machadovilaca) [reply tools/prom-metrics-collector/metrics_collector.go:52]: not really, if it was needed `pull-kubevirt-generate` job would have failed
`kubevirt_vmi_cpu_affinity` was probably not needed there either

**Review by fossedihelm — COMMENTED** [review]
I have a general question: To me, here we are deleting a metric(`kubevirt_vmi_cpu_affinity`) and adding a new one(`kubevirt_vmi_node_cpu_affinity`). Shouldn't we respect some deprecation policy? Thanks @machadovilaca

**Comment by sradco** [comment]
The metrics are not considered `stable` or `beta` at this point.

**Comment by fossedihelm** [comment]
> The metrics are not considered `stable` or `beta` at this point.

Thanks!

**Pull request merged**